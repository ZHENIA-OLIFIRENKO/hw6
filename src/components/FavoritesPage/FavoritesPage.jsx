import React, { useState, useEffect } from 'react';
import Modal from '../Modal/ModalText';
import './FavoritesPage.scss';

const FavoritesPage = () => {
    const [favoritesItems, setFavoritesItems] = useState([]);
    const [removedProduct, setRemovedProduct] = useState(null);
    const [isModalOpen, setIsModalOpen] = useState(false);

    useEffect(() => {
        const storedFavorites = localStorage.getItem('favorites');
        if (storedFavorites) {
            setFavoritesItems(JSON.parse(storedFavorites));
        }
    }, []);

    const removeFromFavorites = (productId) => {
        setRemovedProduct(productId);
        setIsModalOpen(true);
    };

    const handleConfirmRemove = () => {
        const updatedFavorites = favoritesItems.filter((product) => product.article !== removedProduct);
        setFavoritesItems(updatedFavorites);
        localStorage.setItem('favorites', JSON.stringify(updatedFavorites));

        setIsModalOpen(false);
    };

    const handleCancelRemove = () => {
        setIsModalOpen(false);
    };

    return (
        <div className="favorites-page">
            <h2 className="favorites-title">Favorites</h2>

            <ul className="product-list">
                {Array.isArray(favoritesItems) && favoritesItems.length > 0 ? (
                    favoritesItems.map((product) => (
                        <li key={product.article} className="product-item">
                            <div className="product-info">
                                <p className="product-name">{product.name}</p>
                                <p className="product-price">â‚´{(parseFloat(product.price))}</p>
                            </div>
                            <button
                                className="remove-button"
                                onClick={() => removeFromFavorites(product.article)}
                            >
                                Remove
                            </button>
                        </li>
                    ))
                ) : (
                    <li key="empty-favorites" className="empty-favorites">
                        Your favorites list is empty.
                    </li>
                )}
            </ul>

            {/* Modal */}
            <Modal
                isOpen={isModalOpen}
                onClose={handleCancelRemove}
                title="Remove Item"
                content="Are you sure you want to remove this item from favorites?"
                onCancel={handleCancelRemove}
                onConfirm={handleConfirmRemove}
            />
        </div>
    );
};

export default FavoritesPage;
