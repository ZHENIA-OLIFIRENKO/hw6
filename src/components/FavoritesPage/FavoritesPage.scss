.favorites-page {
    padding: 20px;
    background-color: #f4f4f4;
}

.favorites-title {
    font-size: 24px;
    color: #2B435E;
    margin-bottom: 20px;
}

.product-list {
    list-style: none;
    padding: 0;
    margin: 0;
}

.product-item {
    border: 2px dashed rgba($color: #2B435E, $alpha: 0.8);
    border-radius: 5px;
    margin-bottom: 10px;
    padding: 10px;
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.product-info {
    flex: 1;
}

.product-name {
    font-size: 16px;
    color: #2B435E;
}

.product-price {
    font-size: 14px;
    color: #888;
}

.remove-button {
    background-color: #2B435E;
    color: #fff;
    padding: 5px 10px;
    border: none;
    cursor: pointer;
    font-size: 14px;
}

.remove-button:hover {
    background-color: #2b435ee0;
}

.empty-favorites {
    font-size: 16px;
    color: #2B435E;
    margin-top: 20px;
}