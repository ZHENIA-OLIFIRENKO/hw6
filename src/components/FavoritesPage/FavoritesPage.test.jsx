import React from 'react';
import { render } from '@testing-library/react';
import FavoritesPage from './FavoritesPage';

test('рендерить сторінку обраного товару правильно', () => {
    const { container } = render(<FavoritesPage />);
    expect(container).toMatchSnapshot();
});