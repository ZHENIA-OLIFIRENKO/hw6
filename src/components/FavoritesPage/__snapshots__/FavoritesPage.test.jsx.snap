// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders favorite page correctly 1`] = `
<div>
  <div
    class="favorites-page"
  >
    <h2
      class="favorites-title"
    >
      Favorites
    </h2>
    <ul
      class="product-list"
    >
      <li
        class="empty-favorites"
      >
        Your favorites list is empty.
      </li>
    </ul>
  </div>
</div>
`;

exports[`renders favorites page correctly 1`] = `
<div>
  <div
    class="favorites-page"
  >
    <h2
      class="favorites-title"
    >
      Favorites
    </h2>
    <ul
      class="product-list"
    >
      <li
        class="empty-favorites"
      >
        Your favorites list is empty.
      </li>
    </ul>
  </div>
</div>
`;

exports[`рендерить сторінку обраного товару правильно 1`] = `
<div>
  <div
    class="favorites-page"
  >
    <h2
      class="favorites-title"
    >
      Favorites
    </h2>
    <ul
      class="product-list"
    >
      <li
        class="empty-favorites"
      >
        Your favorites list is empty.
      </li>
    </ul>
  </div>
</div>
`;
